common/utils.c

```diff
@@ -416,3 +416,106 @@ TDNFYesOrNo(
     return dwError;
 }
 
+uint32_t TDNFUriIsRemote(
+    const char* pszKeyUrl,
+    int *pnRemote
+)
+{
+    uint32_t dwError = 0;
+    const char *szRemotes[] = {"http://", "https://", "ftp://",
+                               "ftps://", NULL};
+    int i = 0;
+
+    if(!pnRemote || IsNullOrEmptyString(pszKeyUrl))
+    {
+      dwError = ERROR_TDNF_INVALID_PARAMETER;
+      BAIL_ON_TDNF_ERROR(dwError);
+    }
+
+    *pnRemote = 0;
+    for(i = 0; szRemotes[i]; i++) {
+        if (strncasecmp(pszKeyUrl, szRemotes[i], strlen(szRemotes[i])) == 0) {
+            *pnRemote = 1;
+            break;
+        }
+    }
+    if (szRemotes[i] == NULL && strncasecmp(pszKeyUrl, "file://", 7) != 0) {
+       dwError = ERROR_TDNF_KEYURL_INVALID;
+       BAIL_ON_TDNF_ERROR(dwError);
+    }
+cleanup:
+    return dwError;
+
+error:
+    goto cleanup;
+}
+
+uint32_t TDNFFileNameFromUri(
+    const char* pszKeyUrl,
+    char** ppszFile)
+{
+    uint32_t dwError = 0;
+    const char* pszPath = NULL;
+    char *pszFile = NULL;
+    size_t nOffset;
+    const char *szPrefixes[] = {"http://", "https://", "ftp://",
```

Comment:
Call this variable protocols to match the URL terminologies. Also, your error messages must be generic, no key related errors from here.

Reply:
Done. Maybe could have been called `pszSchemes` too.

Reply:
I also addressed the error message, and added a new code. It's a bit clums though because it needs to be translated by the caller.

Reply:
do we have use cases to support http and ftp? @cbsiddharth please check with security on this. i remember discussions around this..

Reply:
@ppadmavilasom, missed this. I don't think we need FTP but we'd need HTTP and HTTPS for sure. Wherever plain HTTP is involved, we must make sure we don't fetch content that needs authentication and this decision must be made on a case-by-case manner. For instance, fetching RPMs from HTTP (or FTP) sources is okay as they are GPG checked later but fetching signature files from such locations is a security gap.

Since this method is a utility method, the callers of this method must go through security audit, IMO.